src/main/java/com/hotels/bdp/circustrain/bigquery/conf/PartitioningConfiguration.java

```diff
@@ -0,0 +1,97 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.bigquery.conf;
+
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+
+import static com.hotels.bdp.circustrain.bigquery.CircusTrainBigQueryConstants.PARTITION_BY;
+import static com.hotels.bdp.circustrain.bigquery.CircusTrainBigQueryConstants.PARTITION_FILTER;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.hotels.bdp.circustrain.api.CircusTrainException;
+import com.hotels.bdp.circustrain.api.conf.SourceTable;
+import com.hotels.bdp.circustrain.api.conf.TableReplication;
+import com.hotels.bdp.circustrain.api.conf.TableReplications;
+import com.hotels.bdp.circustrain.bigquery.util.TableNameFactory;
+
+@Component
+public class PartitioningConfiguration {
+
+  private static final Logger log = LoggerFactory.getLogger(PartitioningConfiguration.class);
+
+  private final Map<String, Map<String, Object>> replicationConfigMap;
+
+  @Autowired
+  PartitioningConfiguration(TableReplications tableReplications) {
+    HashMap<String, Map<String, Object>> replicationConfigMap = new HashMap<>();
+    for (TableReplication tableReplication : tableReplications.getTableReplications()) {
+      SourceTable sourceTable = tableReplication.getSourceTable();
+      String key = TableNameFactory.newInstance(sourceTable.getDatabaseName().trim().toLowerCase(),
+          sourceTable.getTableName().trim().toLowerCase());
+      if (replicationConfigMap.containsKey(key)) {
+        throw new CircusTrainException(
+            "Partitioning cannot be carried out when there are duplicate source tables with partitioning configured");
+      }
+      log.debug("Loading BigQuery partitioning configuration for table {}", key);
+      Map<String, Object> copierOptions = tableReplication.getCopierOptions();
+      replicationConfigMap.put(key, copierOptions);
+    }
+    this.replicationConfigMap = Collections.unmodifiableMap(replicationConfigMap);
+  }
+
+  public String getPartitionFilterFor(Table table) {
+    String key = TableNameFactory.newInstance(table);
+    log.debug("Loading 'partition-filter' for table {}", key);
+    if (replicationConfigMap.containsKey(key)) {
+      Map<String, Object> copierOptions = replicationConfigMap.get(key);
+      if (copierOptions != null && copierOptions.containsKey(PARTITION_FILTER)) {
+        Object yamlValue = copierOptions.get(PARTITION_FILTER);
+        if (yamlValue != null) {
+          return yamlValue.toString();
+        }
+      }
+    }
+    return null;
+  }
+
+  public String getPartitionByFor(Table table) {
+    String key = TableNameFactory.newInstance(table);
```

Comment:
A lot of duplicated code between this and the above method, should be easy to create a new private method which both delegate to, just passing in the key name.